### ISSUE TITLE
Processed/compressed images on build are much larger in size (bytes) than the source

### ISSUE THREAD

**Issue by a-kriya** [gridsome/gridsome#1091]
## Description

Upon auditing my site, I noticed one particular image that was 8 times the size (1.2mB vs 150kB) than what I had actually used. I confirmed that many of the assets that were output in `dist/` by Gridsome on build were much larger in size, probably cause I manually compress images before using them. This increases the load time of the pages by quite a bit, and hurts SEO rankings in turn.

### Steps to reproduce

These steps just outline my situation, so there could be better ways to reproduce the issue.

- Find a fairly large PNG image (~2000x2000) (to see an exaggerated difference)
- Compress it locally or [online](https://compresspng.com/) with best compression levels
- Use the image as a source on a page
- `build` the site and compare the sizes of the output and the source file

### Expected result

All processed images on build *must* be smaller size. If they're not, the source should be used as is.

### Actual result

Notice on the left that the source image's size is 155kB and dimensions are 2800x2900, and the largest (dimension) generated image is still smaller at 2560x2651, but at a whopping 1.2mB (right). Two of the other three versions for smaller viewports are still multiple times in size than the original image.

<img width="1387" alt="GridsomeImages" src="https://user-images.githubusercontent.com/26761352/78839655-87891a80-7a16-11ea-8168-c74eeabd7165.png">

The increase in size can also be noticed above for `sec-2-aside` and `sec-3-aside`, although not as magnified, so this is not an isolated case. If needed, I can provide multiple images that can be used as a source for reproduction.

### Environment

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
  Binaries:
    Node: 12.16.1 - /usr/local/bin/node
    npm: 6.14.2 - /usr/local/bin/npm
  Browsers:
    Firefox: 7x.x
    Safari: 1x.x.x
  npmPackages:
    @gridsome/source-filesystem: ^0.6.2 => 0.6.2
    @gridsome/transformer-remark: ^0.4.0 => 0.4.0
    gridsome: ^0.7.13 => 0.7.13
    gridsome-plugin-gtm: ^0.1.1 => 0.1.1
  npmGlobalPackages:
    @gridsome/cli: 0.3.1
```

**Comment by a-kriya** [comment]
Until this is addressed, does anyone know of a way to disable the built-in compression? Perhaps it could be done using the Server API.

**Comment by noxify** [comment]
Hi @a-kriya,

seems to be a problem only with `png` files.

I have tested it with with a jpg file which has a original size from `2,1 MB` (4096x2733).
After running `gridsome build` the processed images are:

* 2560x1708 - 415kb
* 1024x683 - 72kb
* 480*320 - 19kb
* 1920*1281 - 225kb

To your question:
> Until this is addressed, does anyone know of a way to disable the built-in compression? Perhaps it could be done using the Server API.

Based on the code, it seems not possible to disable this.

But could be solved easily with a new config parameter and some small changes: https://github.com/gridsome/gridsome/blob/master/gridsome/lib/build.js#L32

Hope that helps :)

**Comment by a-kriya** [comment]
Thanks for looking into this, @noxify.

I must've been tired when filing this ticket cause I wrote PNG, but actually picked a JPG for the screenshot to show the issue. So, we've confirmed it exists for JPGs *and* PNGs : )
By the way, since we got different results for JPG, did you manually compress it before using it as a source? Because it seems that the image processors used by Gridsome may not be able to compress the images at ratios that are possible with some other compressors out there.

A config would take care of it, yeah. I also realized that I already bypass processing for some assets in my current project:

![image](https://user-images.githubusercontent.com/26761352/78905102-1b042f00-7a9b-11ea-81a4-44d63161bac9.png)

That's the `static/` dir at the root of the project which [gets copied as is to `dist/` on build](https://gridsome.org/docs/directory-structure/#the-static-directory), so I recreate the structure for where I want the files to go.

**Comment by noxify** [comment]
Hi @a-kriya - no I did not compressed the image - just downloaded it from unsplash and used it .

I will upload my used setup tomorrow.

**Comment by a-kriya** [comment]
@noxify I compressed one of the cover images from your project from 2.1mB to 400kB without noticeable quality loss at http://jpeg-optimizer.com/, and then largest of the generated images at build was at the same size (but smaller dimensions). Source on the left:

![image](https://user-images.githubusercontent.com/26761352/79518299-5d66d680-806e-11ea-8246-4d7c8fbafd79.png)

So, in this case at least the generated image's size does not increase. What you and I suggested above -- a config param to disable image processing, and storing assets in /static/ -- are more like workarounds. Would it be better if Gridsome itself checked after each image is processed if the size has increased, in which case it will instead choose to simply copy over the source image?
